File: src/main/scala/com/kirkwoodwest/scalawoods/utils/FileUtil.scala
package com.kirkwoodwest.scalawoods.utils

import com.bitwig.extension.api.PlatformType
import com.bitwig.extension.controller.api.ControllerHost
import com.bitwig.extension.api.PlatformType.{LINUX, MAC, WINDOWS}
object FileUtil {
  //Static method to get file path
  def getPath(host: ControllerHost, file: String): String = host.getPlatformType match {
    case platform if platform.equals(PlatformType.WINDOWS) =>
      val userProfile = System.getenv("USERPROFILE").replace("\\", "/")
      userProfile + "/Documents/Bitwig Studio/Extensions/" + file
    case platform if platform.equals(PlatformType.MAC) =>
      System.getProperty("user.home") + "/Documents/Bitwig Studio/Extensions/" + file
    case platform if platform.equals(PlatformType.LINUX) =>
      System.getProperty("user.home") + "/Bitwig Studio/Extensions/" + file
    case _ =>
      throw new IllegalArgumentException("Unknown Platform")
  }
}

File: src/main/java/com/kirkwoodwest/openwoods/utils/FileUtil.java
package com.kirkwoodwest.openwoods.utils;

import com.bitwig.extension.controller.api.ControllerHost;

public class FileUtil {
  //Static method to get file path
  public static String getPath(final ControllerHost host, final String file)
  {
    switch (host.getPlatformType())
    {
      case WINDOWS:
        final String userProfile = System.getenv("USERPROFILE").replace("\\", "/");
        return userProfile + "/Documents/Bitwig Studio/Extensions/" + file;

      case MAC:
        return System.getProperty("user.home") + "/Documents/Bitwig Studio/Extensions/" + file;

      case LINUX:
        return System.getProperty("user.home") + "/Bitwig Studio/Extensions/" + file;

      default:
        throw new IllegalArgumentException("Unknown Platform");
    }
  }
}
